File: go/space-age/space_age.go
package space

// Planet basic string type to represent planet names
type Planet string

var secondsPerEarthYear = 31557600.0

var orbitalPeriods = map[Planet]float64{
	"Earth":   secondsPerEarthYear * 1.0,
	"Mercury": secondsPerEarthYear * 0.2408467,
	"Venus":   secondsPerEarthYear * 0.61519726,
	"Mars":    secondsPerEarthYear * 1.8808158,
	"Jupiter": secondsPerEarthYear * 11.862615,
	"Saturn":  secondsPerEarthYear * 29.447498,
	"Uranus":  secondsPerEarthYear * 84.016846,
	"Neptune": secondsPerEarthYear * 164.79132,
}

// Age calculates age
func Age(seconds float64, planet Planet) float64 {
	return seconds / orbitalPeriods[planet]
}

File: c/space-age/space_age.c
#include "space_age.h"

const float orbitalPeriods[8] = {
    0.2408467f,  // MERCURY
    0.61519726f, // VENUS
    31557600.0f, // EARTH
    1.8808158f,  // MARS
    11.862615f,  // JUPITER
    29.447498f,  // SATURN
    84.016846f,  // URANUS
    164.79132f   // NEPTUNE
};

float age(planet_t planet, int64_t seconds)
{
    if (planet != EARTH)
    {
        return seconds / (orbitalPeriods[planet] * orbitalPeriods[EARTH]);
    }
    else
    {
        return seconds / orbitalPeriods[planet];
    }
}
